pkg/library/protobom.go

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright 2025 The Protobom Authors
+
+// Package library defined CEL compile and program options in an object that can
+// be used as a library in a CEL environment.
+package library
+
+import (
+	"github.com/google/cel-go/cel"
+	"github.com/protobom/protobom/pkg/sbom"
+
+	"github.com/protobom/cel/pkg/elements"
+)
+
+const (
+	Name = "cel.protobom.api"
+)
+
+type Protobom struct{}
+
+func NewProtobom() *Protobom {
+	return &Protobom{}
+}
+
+// Types returns the types that the library defines in the CEL environment
+func (p *Protobom) Types() []cel.EnvOption {
+	return []cel.EnvOption{
+		cel.Types(elements.DocumentType),
+		cel.Types(&sbom.Document{}),
+		cel.Types(&sbom.NodeList{}),
+		cel.Types(&sbom.Node{}),
+	}
+}
+
+// Variables defines the global variables that are created in the CEL
+// environment when the library is included
+func (p *Protobom) Variables() []cel.EnvOption {
+	return []cel.EnvOption{
+		cel.Variable("sboms", cel.MapType(cel.IntType, elements.DocumentType)),
+		cel.Variable("protobom", elements.ProtobomType),
+	}
+}
+
+// TypeAdapters wraps the protobom custom type adapter into an option
+// that can be injected into the CEL environment
+func (p *Protobom) TypeAdapters() []cel.EnvOption {
+	return []cel.EnvOption{
+		cel.CustomTypeAdapter(&TypeAdapter{}),
+	}
+}
+
+// createEnvironment creates the CEL execution environment that the runner will
+// use to compile and evaluate programs on the SBOM
+func (p *Protobom) CompileOptions() []cel.EnvOption {
+	ret := []cel.EnvOption{}
+	ret = append(ret, p.Types()...)
+	ret = append(ret, p.Variables()...)
+	ret = append(ret, p.Functions()...)
+	ret = append(ret, p.TypeAdapters()...)
+	return ret
```

Comment:
How do you feel about

```suggestion
	return slices.Concat(p.Types(), p.Variables(), p.Functions(), p.TypeAdapters())
```